### ISSUE TITLE
Status of HPET on Alder Lake-S/Z690 platforms

### ISSUE THREAD

**Issue by zirblazer** [Dasharo/dasharo-issues#95]
During several recent platforms generations, there are inconsistent recommendations about whenever enabling or disabling the HPET based on vendor and platform, and mixed user reports.
On Alder Lake-S, there has been recommendations to just disable HPET since it causes slowdowns or stuttering in Windows systems (Unless you manually change the timer source OS side). In many cases, the Firmware may not give control to enable or disable HPET, which ironically end up causing a lot of BIOS modding requests to add that control back. Since Coreboot relies on a completely different code base, can it be used to test current HPET issues so that we can finally answer the question about whenever these issues are Hardware based or Firmware based?

https://www.overclock.net/threads/msi-pro-z690-a-ddr4-wifi-owners-thread.1795717/page-25#post-28994566
https://winraid.level1techs.com/t/request-msi-z690-edge-ddr4-bios-mod/39870/7
https://www.anandtech.com/show/12678/a-timely-discovery-examining-amd-2nd-gen-ryzen-results/3
https://www.anandtech.com/show/12678/a-timely-discovery-examining-amd-2nd-gen-ryzen-results/4

**Comment by miczyg1** [comment]
Yes we can. We have a switch `USE_PM_ACPI_TIMER` in coreboot build config menu. The thing with HPET is that:
1. It may be a hardware timer like in older generations.
2. it may be emulated in microcode like on modern systems.

Anyways, easy task to test
